src/execution_plan/record.c

```diff
@@ -89,17 +89,37 @@ RecordEntryType Record_GetType(const Record r, int idx) {
 }
 
 SIValue Record_GetScalar(Record r, int idx) {
-	r->entries[idx].type = REC_TYPE_SCALAR;
+	assert(r->entries[idx].type == REC_TYPE_SCALAR);
 	return r->entries[idx].value.s;
 }
 
 Node *Record_GetNode(const Record r, int idx) {
-	r->entries[idx].type = REC_TYPE_NODE;
+	switch(r->entries[idx].type) {
+	case REC_TYPE_NODE:
+		return &(r->entries[idx].value.n);
+	case REC_TYPE_UNKNOWN:
+		return NULL;
+	case REC_TYPE_SCALAR:
+		// Null scalar values are expected here; otherwise fall through.
+		if(SIValue_IsNull(r->entries[idx].value.s)) return NULL;
+	default:
+		assert("encountered unexpected type in Record; expected Node" && false);
```

Comment:
please add the type

Reply:
Since this is a string literal, the only way to do that would be to have separate assertions for each case, which feels like overkill to me. (Even the current case statement layout kind of feels like overkill to me.) Thoughts?